hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/ipc/Client.java

```diff
@@ -392,7 +394,7 @@ private class Connection extends Thread {
     private IOException closeException; // close reason
 
     private final Thread rpcRequestThread;
-    private final SynchronousQueue<Pair<Call, ResponseBuffer>> rpcRequestQueue =
+    private final SynchronousQueue<Pair<Call, Object>> rpcRequestQueue =
```

Comment:
Why should it be replaced by Object?

Reply:
Because for WritableRpcEngine, there is no easy way to calculate the size of request parameters, ResponseBuffer is a better choice, so Object represents ResponseBuffer or RpcProtobufRequestWithHeader.